yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -573,10 +573,16 @@ object ApplicationMaster extends Logging {
   }
 
   private[spark] def sparkContextInitialized(sc: SparkContext): Unit = {
+    if (master == null){
```

Comment:
@vanzin We thought we can't get into this situation, but we are defending against one right now. I will remove it if you will, but I think this helps more than hurts.